### PR TITLE
Fix transaction scope outbox to allow floating the transaction scope to user code

### LINKED ISSUE #504: The new 'UseTransactionScope' option for outbox does not work as expected
We are running NServicebus 7.4.0 with `NServiceBus.Persistence.Sql`  6.0.2. We have recently activated `TransactionScope` as the transaction type for the outbox according [to the NServiceBus documentation](https://docs.particular.net/persistence/sql/outbox#transaction-type-transaction-scope). 

To my surprise, `Transaction.Current` is `null` in my message handler. Here's a sample app that can be used to reproduce the issue: https://github.com/pardahlman/nsb-trasaction-scope-repro

Have I missed something, or is this expected behavior?

### PR DESCRIPTION
**Pull request by SzymonPobiega**
Fixes #504

### CODE REVIEW

File: src/SqlPersistence.Tests/Outbox/OutboxPersisterTests.cs [hunk 1]
```diff
@@ -161,6 +161,26 @@ public void StoreAndGet()
         Approver.Verify(result);
     }
 
+    [Test]
+    public async Task TransactionScope()
```
Comment (tmasternak) [inline src/SqlPersistence.Tests/Outbox/OutboxPersisterTests.cs:165]: Do we need both tests? I think that ATT is good enough.

**Review by tmasternak — DISMISSED** [review]

File: src/SqlPersistence.Tests/Outbox/OutboxPersisterTests.cs (hunk 1 above)
Comment (danielmarbach) [reply src/SqlPersistence.Tests/Outbox/OutboxPersisterTests.cs:165]: I'm in favor of having both layers. One exercising the component behavior and one exercising the user facing behavior

**Review by danielmarbach — APPROVED** [review]

**Review by danielmarbach — APPROVED** [review]

**Review by tmasternak — APPROVED** [review]

**Comment by danielmarbach** [comment]
@SzymonPobiega @tmasternak previously 15/16 tests failed. Can someone look at those?

**Comment by SzymonPobiega** [comment]
@danielmarbach I don't see any failing tests in the historic builds in TC https://builds.particular.net/project.html?projectId=NServiceBusPersistenceSql&branch_NServiceBusPersistenceSql=pull%2F505%2Fmerge

**Comment by larsw** [comment]
When will the new package be available on nuget.org?

**Comment by SzymonPobiega** [comment]
@larsw we're just waiting for the build to complete and we'll be pushing packages

**Comment by tmasternak** [comment]
@larsw we are working on pushing the packages as we speak.

**Pull request merged**